presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFileFooterReader.java

```diff
@@ -48,24 +54,46 @@ public PageFileFooterReader(
         int footerSize = Slices.wrappedBuffer(buffer, buffer.length - FOOTER_LENGTH_IN_BYTES, FOOTER_LENGTH_IN_BYTES).getInt(0);
 
         footerOffset = fileSize - footerSize;
+        HiveCompressionCodec compression;
         if (footerOffset < 0) {
-            throw new IOException("Malformed PageFile format, incorrect footer length.");
+            throw new PageFileCorruptionException("Malformed PageFile format, incorrect footer length.");
         }
         else if (footerOffset > 0) {
```

Comment:
`else if (footerOffset > 0) {` is not needed

Reply:
> `else if (footerOffset > 0) {` is not needed

This is needed because for empty file, ,footerOffset = 0

Reply:
Oh, right. Nevermind then.